src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/hotspot/HotSpotResolvedJavaMethod.java

```diff
@@ -58,6 +58,13 @@ public interface HotSpotResolvedJavaMethod extends ResolvedJavaMethod {
      */
     boolean hasReservedStackAccess();
 
+    /**
+     * Returns true if this method has a {@code Scoped} annotation.
+     *
+     * @return true if Scoped annotation present, false otherwise
+     */
+    boolean isScoped();
```

Comment:
I don't know how much sense it makes to expose a flag that is only relevant for an optimization in HotSpot (and that exposes the presence of an annotation used only internally by the jdk) to non-hotspot methods

Reply:
I think JVMCI is already very much JDK-specific so the question is rather whether it's need by any VM that'd try to interact with this JDK implementation. But i don't want to derail this PR, this can be moved up when/if that's needed.

Reply:
I also agree that this should be moved up to `ResolvedJavaMethod` as the annotation itself is not HotSpot specific and so could easily be relevant for any other VM using the JDK classes (e.g. SVM).

Not that this means you'll need to update `TestResolvedJavaMethod.java` to test the new method but that should be straighforward.

Reply:
I think there's something I'm missing here. I'm trying to extend `TestResolvedJavaMethod.java` to test the new method, but I can't get `isScoped()` to return true for methods with the `@Scoped` annotation such as `jdk.internal.misc.ScopedMemoryAccess#loadInternal()`.

At first I thought it was a problem with my implementation, but I also can't get `HotSpotResolvedJavaMethod#hasReservedStackAccess()` to return true for a method like the following:
```
@jdk.internal.vm.annotation.ReservedStackAccess
private static void methodWithReservedStackAccess() {}
```

The `loadInternal()` method above also has a `ForceInline` annotation, and for that one `HotSpotResolvedJavaMethod#isForceInline()` does return true.

The only difference I can see between these flags is that `isForceInline()` is stored in `Method::_flags` whereas `loadInternal()` and `hasReservedStackAccess()` are stored in `ConstMethod::_flags`. Could it be that const method flags are not being propagated properly in tests (or elsewhere)?

Reply:
Nevermind, looks like the issue with `hasReservedStackAccess` was the fact that the holding class has to be loaded by the boot class loader for the annotation to be picked up. Back to the drawing board...

Reply:
FYI those internal annotations are meaningful only for boot/platfrom-loaded classes as well as hidden classes loaded with the `java.lang.invoke.MethodHandleNatives.Constants#ACCESS_VM_ANNOTATIONS` flag. See https://github.com/openjdk/jdk/blob/7f11935461a6ff1ef0fac800fb4264a519e21061/src/hotspot/share/classfile/classFileParser.cpp#L1916-L1919.
The javadoc that just says that it's about the method having this annotation is a bit misleading.